src/PowerShellEditorServices/Services/DebugAdapter/Handlers/LaunchAndAttachHandler.cs

```diff
@@ -250,11 +284,13 @@ public async Task<AttachResponse> Handle(PsesAttachRequestArguments request, Can
             _debugService.IsDebuggingRemoteRunspace = true;
             try
             {
+                _debugService.SetPathMappings(request.PathMappings);
                 return await HandleImpl(request, cancellationToken).ConfigureAwait(false);
             }
             catch
             {
                 _debugService.IsDebuggingRemoteRunspace = false;
+                _debugService.UnsetPathMappings();
                 throw;
```

Comment:
Ohh I see where that pattern came from now.

Reply:
Yea it's done in the catch block to undo any setup work for the session in case of a fatal failure. The normal cleanup will happen in the terminal/disconnect handler which occurs when the debug session ends.